File: src/main/java/com/aleprimo/Booking_System_App/service/UserService.java
package com.aleprimo.Booking_System_App.service;


import com.aleprimo.Booking_System_App.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface UserService {
    User createUser(User user);
    User updateUser(Long id, User user);
    void deleteUser(Long id);
    Optional<User> getUserById(Long id);
    Optional<User> getUserByEmail(String email);
    Page<User> getAllUsers(Pageable pageable);
}

File: booking-frontend/src/api/userService.js
import api from "./axiosClient";

export const getUsers = (page = 0, size = 10, token) =>
  api.get(`/api/users?page=${page}&size=${size}`, {
    headers: { Authorization: `Bearer ${token}` },
  });

export const getUserById = (id, token) =>
  api.get(`/api/users/${id}`, {
    headers: { Authorization: `Bearer ${token}` },
  });

export const updateUser = (id, data, token) =>
  api.put(`/api/users/${id}`, data, {
    headers: { Authorization: `Bearer ${token}` },
  });

export const deleteUser = (id, token) =>
  api.delete(`/api/users/${id}`, {
    headers: { Authorization: `Bearer ${token}` },
  });
export const getUserByEmail = (email, token) =>
  api.get(`/api/users/email/${email}`, {
    headers: { Authorization: `Bearer ${token}` },
  });